src/main/java/codechicken/multipart/trait/TSlottedTile.java

```diff
@@ -51,17 +51,19 @@ public void partRemoved(MultiPart part, int p) {
     }
 
     @Override
-    public boolean canAddPart(MultiPart part) {
-        if (part instanceof SlottedPart) {
-            int mask = ((SlottedPart) part).getSlotMask();
+    public boolean canAddParts(Iterable<MultiPart> parts) {
```

Comment:
We probably need to validate that the slotted parts in this list don't overlap. I.e Someone tries to add 2 parts in the same slot.

Reply:
Done. Now, we first build combined mask of all new parts, ensuring no overlap. Then we check that against the existing part list.